src/frontend/src/pages/GanttPage/GanttPageWrapper.tsx

```diff
@@ -62,16 +63,159 @@ const GanttPageWrapper: React.FC = () => {
     };
   };
 
-  const projTasks = projects.map(transformProjectToTask);
+  useEffect(() => {
+    const transformProjectToGanttObject = (project: Project): Task => {
+      return {
+        id: wbsPipe(project.wbsNum),
+        name: wbsPipe(project.wbsNum) + ' ' + project.name,
+        start: project.startDate || new Date(),
+        end: project.endDate || new Date(),
+        progress:
+          (project.workPackages.filter((wp) => wp.status === WbsElementStatus.Complete).length /
+            project.workPackages.length) *
+          100,
+        type: 'project',
+        hideChildren: !expanded,
+        styles: { progressColor: '#e50000', backgroundColor: '#ff0000' },
+        displayOrder: project.id,
+        onClick: () => {
+          window.open(`/projects/${wbsPipe(project.wbsNum)}`, '_blank');
+        }
+      };
+    };
```

Comment:
can you put this transformProjectToGanttObject method outside the useEffect and just call it when you need it

Reply:
Causes infinite loop, eslint told me to me put inside.